### PR TITLE
feat: Add support for extra fields on Person details.

### PR DESCRIPTION
**Pull request by elliotcourant**
## Description of the change

This is to help achieve parody with the PHP SDK which allows for
additional fields to be specified on the Person object. We use this to
provide details like what account a specific user belongs to or other
additional internal metadata that is helpful when debugging user/account
level errors.

This is mainly mirroring what I saw in the PHP SDK here: 

https://github.com/rollbar/rollbar-php/blob/89c9069e8532acb638f2fb9efd36b529ec25354f/src/Payload/Person.php#L68-L80

```php
    public function serialize()
    {
        $result = array(
            "id" => $this->id,
            "username" => $this->username,
            "email" => $this->email,
        );
        foreach ($this->extra as $key => $val) {
            $result[$key] = $val;
        }
        
        return $this->utilities()->serializeForRollbarInternal($result, array_keys($this->extra));
    }
```

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Maintenance
- [ ] New release

## Related issues

From what I've seen there are not any GitHub issues for this.

### Development

- [ ] Lint rules pass locally
      _(I don't see any editor config or linting documentation in the README or anywhere else in the repo?)_
- [x] The code changed/added as part of this pull request has been covered with tests
      _(This code should be covered by existing unit tests, but happy to add a specific test for this code path)_
- [ ] All tests related to the changed code pass in development

### Code review

- [x] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers assigned
- [ ] Issue from task tracker has a link to this pull request
- [ ] Changes have been reviewed by at least one other engineer

### CODE REVIEW

File: transforms.go [hunk 1]
```diff
@@ -50,11 +50,21 @@ func buildBody(ctx context.Context, configuration configuration, diagnostic diag
...
+		personData := map[string]string{
 			"id":       person.Id,
 			"username": person.Username,
 			"email":    person.Email,
 		}
+		for key, value := range person.Extra {
+			// If the field on the extra map is already specified then skip it.
+			// This will prevent the extra map from overwriting fields like ID, Username or Email.
+			if _, ok := personData[key]; ok {
+				continue
```
Comment (pawelsz-rb) [inline transforms.go:62]: I would suggest removing 'continue' and checking for '!ok' 

Also some unit tests would be useful.

**Review by pawelsz-rb — CHANGES_REQUESTED** [review]

File: transforms.go (hunk 1 above)
Comment (ECourant) [reply transforms.go:62]: I'm not gonna write tests for a code base that requires an API key. I'm also no longer at the company that was using this SDK.

If you wanna make these changes go for it; but I'll probably just close the PR. It's been more than 6 months.

Comment (elliotcourant) [reply transforms.go:62]: I'm not gonna write tests for a code base that requires an API key. I'm also no longer at the company that was using this SDK.

If you wanna make these changes go for it; but I'll probably just close the PR. It's been more than 6 months.

**Comment by pawelsz-rb** [comment]
implementation will be here: https://github.com/rollbar/rollbar-go/pull/96

**Pull request closed without merging**